runtime/vm/stringhelpers.cpp

```diff
@@ -498,7 +498,9 @@ catUtfToString4(J9VMThread * vmThread, const U_8 *data1, UDATA length1, const U_
 	j9object_t result = NULL;
 	UDATA totalLength = length1 + length2 + length3 + length4;
 	U_8 *buffer = (U_8*)j9mem_allocate_memory(totalLength, OMRMEM_CATEGORY_VM);
-	if (NULL != buffer) {
+	if (NULL == buffer) {
+		vmThread->javaVM->internalVMFunctions->setNativeOutOfMemoryError(vmThread, 0, 0);
```

Comment:
We're in the VM here.  Can we directly call `setNativeOOM?`?
```suggestion
		setNativeOutOfMemoryError(vmThread, 0, 0);
```

Reply:
The file is recompiled for one of the tests, so the direct call causes a link error.

Reply:
I'd suggest the `J9_VM_FUNCTION(currentThread, function)` approach but it isn't worth the effort for an OOM case

Reply:
That macro only works if the magic ifdef is defined, which it is not for the test directory.